File: my_cpp_pkg/src/robot_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp" 

using namespace std::chrono_literals;

class RobotPubNode : public rclcpp::Node // MODIFY NAME
{
public:
    // characteristics of the node
    RobotPubNode() : Node("robot_publisher") // MODIFY NAME
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("News", 10); // MODIFY TOPIC NAME
        timer_= this->create_wall_timer(0.5s, std::bind(&RobotPubNode::publish_news, this)); // MODIFY TIME
        RCLCPP_INFO(this->get_logger(), "Robot Publisher has been started");
    }
 
private:
    void publish_news()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = "Hello, world!";
        publisher_->publish(msg);
    }
    //declaring variables
   rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
   rclcpp::TimerBase::SharedPtr timer_; 
};
 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotPubNode>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/robot_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

     
     
class Newstation(Node): 
    def __init__(self):
        super().__init__("robot_publisher") # set node name
        self.declare_parameter("robot_name", "Rob")
        self.robot_name = self.get_parameter("robot_name").value # get parameter value
        self.publisher_= self.create_publisher(String, "News", 10) # create publisher for strings on the /News topic
        self.timer = self.create_timer(0.5, self.publish_news) # set timer between every new publish
        self.get_logger().info("Publisher started")

    def publish_news(self):
        msg = String() # create message and set its type
        msg.data = f"Hello world, Im {self.robot_name}" # set message data
        self.publisher_.publish(msg) # publish the entire message
        #self.get_logger().info(f"Publishing: {msg.data}") #publish message with shine
     
     
def main(args=None):
    rclpy.init(args=args)
    node = Newstation()
    rclpy.spin(node)
    rclpy.shutdown()
     
     
if __name__ == "__main__":
    main()
